src/components/FlexibleForm/Control.js

```diff
@@ -287,20 +295,21 @@ export const Control = memo(props => {
     const currentSection = commonSections.find(item => item.sectionKey === acceptedSections[stateFieldValue]);
 
     if (!currentSection?.controlType) {
-      return <>&nbsp;</>;
+      return <></>;
     }
 
     const {
-      childControls: cc,
+      childControls: cc = [],
       optional: isOptional,
     } = currentSection;
 
-    const sectionChildren = isOptional && !isEditable && !hasContent(cc, record, sectionNS, ri) ? [] : cc;
+    const isSectionComponent = currentSection.controlType === 'Section';
+    const sectionChildren = isSectionComponent && isOptional && !isEditable && !hasContent(cc, record, sectionNS, ri) ? [] : cc;
```

Comment:
Please cover `isSectionComponent && isOptional && !isEditable && !hasContent(cc, record, sectionNS, ri)` in other variable